lib/middleman/s3_sync.rb

```diff
@@ -58,12 +58,22 @@ def update_bucket_versioning
       end
 
       def connection
-        @connection ||= Fog::Storage::AWS.new({
-          :aws_access_key_id => s3_sync_options.aws_access_key_id,
-          :aws_secret_access_key => s3_sync_options.aws_secret_access_key,
+
+        connection_options = {
           :region => s3_sync_options.region,
           :path_style => s3_sync_options.path_style
-        })
+        }
+
+        if s3_sync_options.aws_access_key_id && s3_sync_options.aws_secret_access_key
```

Comment:
I think that we need to flip the conditions. We can create a `use_iam_profile` option that would configure fog to use the IAM profile and otherwise fall back to the old authentication approach.

That said, detecting which credentials are available (whether it is via the IAM profile, specified options or through environment variables) is probably the right path to take.